File: flutter/library/lib/src/tsp_card.dart
import 'package:flutter/material.dart';
import 'star_planet_theme.dart';

class TspCard extends StatelessWidget {
  const TspCard({super.key, required this.child, this.selected = false, this.disabled = false, this.theme = StarPlanetTheme.sky});
  final Widget child;
  final bool selected;
  final bool disabled;
  final StarPlanetTheme theme;

  @override
  Widget build(BuildContext context) {
    return Opacity(
      opacity: disabled ? .45 : 1,
      child: Container(
        width: double.infinity,
        padding: const EdgeInsets.all(20),
        decoration: BoxDecoration(
          color: selected ? theme.selectedFill : theme.surfaceRaised,
          borderRadius: BorderRadius.circular(28),
          border: Border.all(color: selected ? theme.success : theme.borderDefault),
        ),
        child: child,
      ),
    );
  }
}

File: ios-swiftui/library/Sources/PlanetComponents/TspCard.swift
import SwiftUI

public struct TspCard<Content: View>: View {
    let selected: Bool
    let disabled: Bool
    let theme: StarPlanetTheme
    let content: Content

    public init(
        selected: Bool = false,
        disabled: Bool = false,
        theme: StarPlanetTheme = .sky,
        @ViewBuilder content: () -> Content
    ) {
        self.selected = selected
        self.disabled = disabled
        self.theme = theme
        self.content = content()
    }

    public var body: some View {
        content
            .padding(20)
            .frame(maxWidth: .infinity, alignment: .leading)
            .background(selected ? theme.selectedFill : theme.surfaceRaised)
            .overlay(
                RoundedRectangle(cornerRadius: 28)
                    .stroke(selected ? theme.success : theme.borderDefault, lineWidth: 1)
            )
            .clipShape(RoundedRectangle(cornerRadius: 28))
            .opacity(disabled ? 0.45 : 1)
    }
}
